ui/src/main/webapp/wizards/sources/validation.spec.js

```diff
@@ -251,4 +252,43 @@ describe('Sources Validation', () => {
       }), 'Next button should be enabled').to.equal(false)
     })
   })
+
+  describe('Valid URLs', () => {
+    const runs = [
+        {it: 'http://foo.com/blah_blah'},
+        {it: 'http://www.example.com/wpstyle/?p=364'},
+        {it: 'http://142.42.1.1/'},
+        {it: 'http://code.google.com/events/#&product=browser'},
+        {it: 'http://j.mp'},
+        {it: 'http://localhost:8993'},
+        {it: 'http://foo.bar/?q=Test%20URL-encoded%20stuff'},
+        {it: 'https://localhost:8993'}
+    ]
+
+    runs.forEach((run) => {
+      it(run.it + ' is valid URL', () => {
```

Comment:
I think creating a new spec for each case just makes the test output noisy. An alternative approach would be creating a single spec, with multiple assertions.

```javascript
it('should be a valid url', () => {
  runs.forEach((run) => {
    // same assertion
  })
})
```